core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -218,41 +219,125 @@ class SnappyContext protected[spark](
       SaveMode.ErrorIfExists, samplingOptions,
       onlyBuiltIn = true, onlyExternal = false)
     DataFrame(self, plan)
   }
 
   /**
+   * Create a stratified sample table. Java friendly version.
+   * @todo provide lot more details and examples to explain creating and
+   *       using sample tables with time series and otherwise
+   * @param tableName the qualified name of the table
+   * @param schema
+   * @param samplingOptions
+   */
+  def createSampleTable(tableName: String, schema: StructType
+      , samplingOptions: java.util.Map[String, String]): DataFrame = {
+    createSampleTable(tableName, schema, samplingOptions.asScala.toMap)
+  }
+
+
+  /**
+   * Create approximate structure to query top-K with time series support.
+   * @todo provide lot more details and examples to explain creating and
+   *       using TopK with time series
+   * @param topKName the qualified name of the top-K structure
+   * @param keyColumnName
+   * @param inputDataSchema
+   * @param topkOptions
+   */
+  def createApproxTSTopK(topKName: String, keyColumnName: String,
+      inputDataSchema: StructType, topkOptions: Map[String, String]): DataFrame = {
+    val plan = createTable(catalog.newQualifiedTableName(topKName),
+      SnappyContext.TOPK_SOURCE, Some(inputDataSchema), schemaDDL = None,
+      SaveMode.ErrorIfExists, topkOptions + ("key" -> keyColumnName),
+      onlyBuiltIn = true, onlyExternal = false)
+    DataFrame(self, plan)
+  }
+
+  /**
    * Create approximate structure to query top-K with time series support.
+   * Java friendly api.
    * @todo provide lot more details and examples to explain creating and
    *       using TopK with time series
    * @param topKName the qualified name of the top-K structure
    * @param keyColumnName
    * @param inputDataSchema
    * @param topkOptions
-   * @param ifExists
    */
   def createApproxTSTopK(topKName: String, keyColumnName: String,
-      inputDataSchema: Option[StructType], topkOptions: Map[String, String],
-      ifExists: Boolean = false): DataFrame = {
+      inputDataSchema: StructType, topkOptions: java.util.Map[String, String]): DataFrame = {
+    createApproxTSTopK(topKName, keyColumnName, inputDataSchema, topkOptions.asScala.toMap)
+  }
+
+  /**
+   * Create approximate structure to query top-K with time series support.
+   * @todo provide lot more details and examples to explain creating and
+   *       using TopK with time series
+   * @param topKName the qualified name of the top-K structure
+   * @param keyColumnName
+   * @param topkOptions
+   */
+  def createApproxTSTopK(topKName: String, keyColumnName: String,
+      topkOptions: Map[String, String]): DataFrame = {
     val plan = createTable(catalog.newQualifiedTableName(topKName),
-      SnappyContext.TOPK_SOURCE, inputDataSchema, schemaDDL = None,
+      SnappyContext.TOPK_SOURCE, None, schemaDDL = None,
       SaveMode.ErrorIfExists, topkOptions + ("key" -> keyColumnName),
```

Comment:
See the comment before about adding back ifExists.

Why inputDataSchema has been removed? I think it was being used for the case when there is no "baseTable" (and user may want to insert DFs explicitly)